sources/packages/web/src/views/aest/institution/DesignationAESTView.vue

```diff
@@ -0,0 +1,221 @@
+    const showModal = ref(false);
+    const toast = useToastMessage();
+    const showActionButtons = computed(
+      () =>
+        designationFormModel.designationStatus !==
+        DesignationAgreementStatus.Declined,
+    );
+    const approveDenyDesignationModal = ref(
+      {} as ModalDialog<UpdateDesignationDto | boolean>,
+    );
+    const updateDesignationModel = ref({} as UpdateDesignationDto);
+    let navigationTitle = "Pending designations";
+    let routeLocation: RouteLocationRaw = {
+      name: AESTRoutesConst.PENDING_DESIGNATIONS,
+    };
+    if (props.institutionId) {
+      navigationTitle = "Manage designations";
+      routeLocation = {
+        name: AESTRoutesConst.INSTITUTION_DESIGNATION,
+        params: { institutionId: props.institutionId },
+      };
+    }
+
+    const loadDesignation = async () => {
+      designationAgreement.value = await DesignationAgreementService.shared.getDesignationAgreement(
+        props.designationAgreementId,
+      );
+
+      designationFormModel.institutionName =
+        designationAgreement.value.institutionName;
+      designationFormModel.institutionType =
+        designationAgreement.value.institutionType;
+      designationFormModel.isBCPrivate = designationAgreement.value.isBCPrivate;
+      designationFormModel.viewMode = DesignationFormViewModes.viewOnly;
+      designationFormModel.designationStatus =
+        designationAgreement.value.designationStatus;
+      designationFormModel.designationStatusClass = mapDesignationChipStatus(
+        designationAgreement.value.designationStatus,
+      );
+      designationFormModel.dynamicData =
+        designationAgreement.value.submittedData;
+      designationFormModel.locations = designationAgreement.value.locationsDesignations.map(
+        location => ({
+          locationId: location.locationId,
+          locationName: location.locationName,
+          requestForDesignation: location.requested,
+          approvedForDesignation: location.approved,
+          locationAddress: formatter.getFormattedAddress({
+            ...location.locationData.address,
+            provinceState: location.locationData.address.province,
+          }),
+        }),
+      );
+    };
+    onMounted(async () => {
+      await loadDesignation();
+    });
+
+    const updateDesignation = async (
+      designationStatus: DesignationAgreementStatus,
+    ) => {
+      if (designationStatus === DesignationAgreementStatus.Approved) {
+        /*If the update action is Approval, the build the designation location array for form
+          by merging the institution locations list with designation locations list
+        */
+        const institutionLocations = await InstitutionService.shared.getAllInstitutionLocationSummary(
+          designationAgreement.value.institutionId,
+        );
+        updateDesignationModel.value.startDate =
+          designationAgreement.value.startDate;
```

Comment:
do we need values that were not updated?

Reply:
Comments updated explaining this.